str_analysis/check_combined_results_tsv_for_pathogenic_repeats.py

```diff
@@ -0,0 +1,274 @@
+                except ValueError as e:
+                    print(f"WARNING: {locus_id} couldn't parse NormalMax fields from disease_info {disease_info}: {e}")
+
+                try:
+                    pathogenic_threshold_min = min(int(float(d["PathogenicMin"])) for d in disease_info if d.get("PathogenicMin"))
+                except ValueError as e:
+                    print(f"WARNING: {locus_id} couldn't parse PathogenicMin fields from disease_info {disease_info}: {e}")
+
+            reference_region = first_row["ReferenceRegion"]
+            genome_version = f"GRCh{first_row['Sample_genome_version']}" if first_row.get('Sample_genome_version') else ""
+            motif = first_row.get("RepeatUnit")
+            locus_description = f"{locus_id} ({reference_region}: {genome_version})  https://stripy.org/database/{locus_id}"
+            print(f"**Locus**: ", locus_description)
+            print("**Disease**: ", str(disease_info))
+            print("**Inheritance**: ", inheritance_mode)
+            print("**Pathogenic Threshold**: >=", pathogenic_threshold_min, "x", motif)
+            print("**Intermediate Threshold**: >=", intermediate_threshold_min, "x", motif)
+
+            locus_specific_df.loc[:, "Sample_sample_type"] = locus_specific_df["Sample_sample_type"].fillna("Unknown")
+            locus_specific_df.loc[:, "Sample_sex"] = locus_specific_df["Sample_sex"].fillna("Unknown")
+            locus_specific_df.loc[:, "Sample_affected"] = locus_specific_df["Sample_affected"].fillna("Unknown")
+
+            filtered_df_list = []
+            for sample_type in sorted(set(locus_specific_df["Sample_sample_type"])):
+                locus_and_sample_type_specific_df = locus_specific_df[
+                    locus_specific_df["Sample_sample_type"].str.contains(sample_type) |
+                    locus_specific_df["Sample_sample_type"].str.contains("Unknown")
+                ]
+
+                if locus_id == "COMP":
+                    # COMP is a special case where contractions below 5 repeats are also pathogenic
+                    df_comp = locus_and_sample_type_specific_df[
+                        (locus_and_sample_type_specific_df["CI end: Allele 1"] < 5) |
+                        (locus_and_sample_type_specific_df["CI end: Allele 2"] < 5)
+                    ]
+                    filtered_df_list.append(df_comp)
+
+                if args.use_thresholds:
+                    threshold = intermediate_threshold_min or pathogenic_threshold_min
+                else:
+                    threshold = 0
+
+                if threshold:
+                    if inheritance_mode == "XR":
+                        df_female = locus_and_sample_type_specific_df[
+                            (locus_and_sample_type_specific_df["Sample_sex"].str.upper().str.startswith("F") | locus_and_sample_type_specific_df["Sample_sex"].str.upper().str.startswith("U")) &
```

Comment:
will sample sex ever be noted with XX/XY instead of male/female?

Reply:
I don't think so, but I should add a check to make sure it's only the values I expect.